README.md

```diff
@@ -1,39 +1,60 @@
 
-Transifex Python Toolkit
-------------------------
+# Transifex Python Toolkit
 
-Transifex Python Toolkit is a collection of tools that allow you to easily localize your applications using Transifex.
+Transifex Python Toolkit (referred as **the Toolkit** for brevity) is a collection of tools that allow you to easily localize your applications using Transifex (TX). The toolkit features fetching translations over the air (OTA) to your apps.
+
+Using Transifex (TX) the localization workflow typically consists of the following steps:
+1. Find - extract the source strings in your source code
+2. Push them into a Transifex project
+3. In Transifex your source strings get translated by translators in target languages
+4. Retrieve a string translation of a target language from Transifex to display it
+
+Transifex Python Toolkit is used in steps 1, 2, 4 above. Possible usage scenarios:
+* [In a Django web application](#scenario-1-integrate-with-a-django-app) The toolkit is fully integrated in Django framework. It contains utilities to mark your source strings for translation and display translation in templates or views. It also provides special CLI commands for pushing to TX and even migrating your existing template code to the new syntax.
+* [In a Python application](#scenario-2-use-transifex-native-as-a-python-library) In other python applications or frameworks you can utilize the toolkit as a python library. The building blocks are there: you can push your source strings to TX and retrieve translations over the air to display them to your users.
+
+# Generic Features - Supported Functionality
+This section refers to the generic functionality provided by the Toolkit. For implementation details refer to the specific usage scenario detailed in following sections.
+
+- **ICU message Format** (variables, plurals), context, metadata (comment, charlimit, tags) support
+- HTML escaping & un-escaping
+- Automatic fetch of translations over-the-air (OTA) via a background thread
+- Policies to handle missing translations or errors
+- Use of an intermediary service, [CDS](#hosting-translations-on-your-servers),to retrieve translations and an in-toolkit memory cache to serve them fast.
 
 # Installation
+Install the Toolkit to your project
 ```python
 pip install transifex-python
 ```
 
-The toolkit provides a Python interface for preparing your code for localization, as well as a thin Django wrapper for Django apps.
+# Transifex Setup
+Before you begin using the Toolkit, you will also need an account in [Transifex](https://www.transifex.com) and a project.
+To set a project compatible with this toolkit contact [support](https://www.transifex.com/contact/) and you
+will be given a set of credentials (a public token and a secret), that you can use in your code for authentication.
+We will refer to these credentials in the text below as:
```

Comment:
Since those are only referenced in the plain python section, I'd remove this note here.

Reply:
They are also referenced in the Setup Django section. That's why I extracted them in a common  setup section. I also made minor amendments to in the Python lib examples to reference more explicitly the values of these credentials.

Reply:
Ah. Right. Okay then :)